src/openmrs-esm-error-handling.ts

```diff
@@ -14,3 +14,47 @@ window.onerror = function() {
 
   return false;
 };
+
+window.onunhandledrejection = function(event) {
+  showToast({ description: `UNHANDLED PROMISE REJECTION: ${event.reason}` });
+
+  return false;
+};
+
+export function reportError(err) {
+  const error = ensureErrorObject(err);
+  setTimeout(() => {
+    throw error;
+  });
+}
+
+export function createErrorHandler() {
+  const outgoingErr = Error();
+  return incomingErr => {
+    const finalErr = ensureErrorObject(incomingErr);
+    finalErr.stack += `\nAsync stacktrace:\n${outgoingErr.stack}`;
+    reportError(incomingErr);
+  };
+}
+
+function ensureErrorObject(thing) {
+  let message;
+  if (thing instanceof Error) {
+    return thing;
+  } else if (typeof thing === "object") {
+    try {
+      message = `Object thrown as error: ${JSON.stringify(thing)}`;
+    } catch (e) {
+      message = `Object thrown as error with the following properties: ${Object.keys(
+        thing
+      )}`;
+    }
+    return Error(message);
+  } else if (thing === null) {
```

Comment:
`typeof null === 'object'` evaluates to `true` (very weird, but true 😄). As such, this if statement will never be hit and is dead code.

To fix it, please move the `else if (thing === null)` to be before the `else if (typeof thing === 'object')`

Reply:
Please implement this ^